src/Storage/Controllers/DataLockController.cs

```diff
@@ -0,0 +1,133 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Altinn.Platform.Storage.Authorization;
+using Altinn.Platform.Storage.Helpers;
+using Altinn.Platform.Storage.Interface.Models;
+using Altinn.Platform.Storage.Repository;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.Cosmos;
+
+namespace Altinn.Platform.Storage.Controllers;
+
+/// <summary>
+/// API for handling locking and unlocking of data elements
+/// </summary>
+[Route("storage/api/v1/instances/{instanceOwnerPartyId:int}/{instanceGuid:guid}/data/{dataGuid:guid}/lock")]
+[ApiController]
+public class DataLockController : ControllerBase
+{
+    private readonly IInstanceRepository _instanceRepository;
+    private readonly IDataRepository _dataRepository;
+    private readonly IAuthorization _authorizationService;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DataLockController"/> class
+    /// </summary>
+    /// <param name="instanceRepository">the instance repository</param>
+    /// <param name="dataRepository">the data repository handler</param>
+    /// <param name="authorizationService">the authorization service.</param>
+    public DataLockController(
+        IInstanceRepository instanceRepository,
+        IDataRepository dataRepository,
+        IAuthorization authorizationService)
+    {
+        _instanceRepository = instanceRepository;
+        _dataRepository = dataRepository;
+        _authorizationService = authorizationService;
+    }
+
+    /// <summary>
+    /// Locks a data element
+    /// </summary>
+    /// <param name="instanceOwnerPartyId">The party id of the instance owner.</param>
+    /// <param name="instanceGuid">The id of the instance that the data element is associated with.</param>
+    /// <param name="dataGuid">The id of the data element to delete.</param>
+    /// <returns></returns>
+    [Authorize(Policy = AuthzConstants.POLICY_INSTANCE_WRITE)]
+    [HttpPut]
```

Comment:
If "lock" is the REST resource then this should be a POST.

Reply:
Agreed, was so focused on the dataelemt part that I forgot that we actually are creating/deleting a lock.

Reply:
As we discussed, left the method as PUT but it returns 200 if already locked and 201 if not previously locked

Reply:
We decided to keep PUT as it is okay to use it to create a resource and is assumed to be idempotent.